OpenRA.Mods.RA2/Traits/ChronoDelivery.cs

```diff
@@ -0,0 +1,105 @@
+#region Copyright & License Information
+/*
+ * Copyright 2007-2015 The OpenRA Developers (see AUTHORS)
+ * This file is part of OpenRA, which is free software. It is made
+ * available to you under the terms of the GNU General Public License
+ * as published by the Free Software Foundation. For more information,
+ * see COPYING.
+ */
+#endregion
+
+using System;
+using OpenRA.Mods.Common.Activities;
+using OpenRA.Mods.Common.Traits;
+using OpenRA.Activities;
+using OpenRA.Traits;
+
+namespace OpenRA.Mods.RA2.Traits
+{
+	public class ChronoDeliveryInfo : ITraitInfo, Requires<HarvesterInfo>
+	{
+		[Desc("The number of ticks between each check to see if we can teleport home.")]
+		public readonly int CheckTeleportDelay = 10;
+
+		public virtual object Create(ActorInitializer init) { return new ChronoDelivery(init.Self, this); }
+	}
+
+	public class ChronoDelivery : INotifyHarvesterAction, ITick
+	{
+		public readonly ChronoDeliveryInfo Info;
+
+		CPos? dest = null;
+		Activity nextActivity = null;
+		int ticksTillCheck = 0;
+
+		public ChronoDelivery(Actor self, ChronoDeliveryInfo info)
+		{
+			Info = info;
+		}
+
+		public void Tick(Actor self)
+		{
+			if (dest == null)
+				return;
+
+			if (ticksTillCheck <= 0)
+			{ 
+			    ticksTillCheck = Info.CheckTeleportDelay;
+
+				CheckTeleport(self);
+			}
+			else
+				ticksTillCheck--;
+		}
+
+		public void MovingToResources(Actor self, CPos targetCell, Activity next)
+		{
+			Reset();
+		}
+
+		public void MovingToRefinery(Actor self, CPos targetCell, Activity next)
+		{
+			if (dest != null && dest.Value != targetCell)
+				ticksTillCheck = 0;
+
+			dest = targetCell;
+			nextActivity = next;
+		}
+
+		public void MovementCancelled(Actor self)
+		{
+			Reset();
+		}
+
+		public void Harvested(Actor self, ResourceType resource) { }
+		public void Docked() { }
+		public void Undocked() { }
+
+		void CheckTeleport(Actor self)
```

Comment:
Could you rename this method to `TeleportIfPossible`?
I would expect a method whose name beings with `Check` to return `bool` and be pure.